mapbox/dr-ui
# Load Search with a facade

**This branch will merge into #431** 

This PR replaces the initial entry point to the Search component with a facade. The facade is a button that looks exactly like the Search component, but only in looks. Once the users clicks or focuses on the facade, the full Search component will load via dynamic import. 

A few things were moved around to support the facade:
1. The logic to load the Swiftype moved to a new file: `search-provider.js`. When the user interacts with the facade, we dynamically load the `search-provider.js`.
2. The search input and search button are now components that are shared by `search-box.js` and `search-facade.js`. This allows us to have a seamless transition.

## How to test

Overall: you should not be able to tell that the Search component loads a facade until you click or focus on it.

1. `npm start` and visit the /Search test cases app.
    - [ ] On small screens, test cases the use the search button switch to input
    - [ ] Each test case returns a search result
2. `npm run start-docs`
    - [ ] Search component works in PageLayout

## QA checklist



- [x] Accessibility score in [Chrome DevTools Audit/Lighthouse](https://developers.google.com/web/tools/lighthouse#devtools) is 100% with Lighthouse version: `7.0.0`.

Open the test cases app locally on:


## Before merge
